Workspace Editor: Splitting Contents and Config

chynu opened the pull request:
Workspace contents and config are now explicitly separate in devtools. Users can only edit one at a time in an editor (e.g. if a user clicks on a config object in the navtree, the contents editor will be disabled but still render a preview of the last contents object the user edited).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly-devtools/236)
<!-- Reviewable:end -->

chynu committed:
Disables other wksp component when editing one

chynu committed:
Can switch, also fixed a bug

chynu committed:
Nits and fixes

chynu commented:
Addresses some of the problems that #226 resolves fully.

AnmAtAnm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> after updating TODO comments.

---

Review status: 0 of 13 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/controller/workspace_controller.js, line 240 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/236#-KrBvtuGfvybkdn6IOYE:-KrBvtuGfvybkdn6IOYF:b-3es077) ([raw file](https://github.com/google/blockly-devtools/blob/893d5c46f61fab62eeabe1f3adf19913de31012f/src/controller/workspace_controller.js#L240)):*
> ```JavaScript
>     this.writeOptions_(options);
>     this.updateOptions();
>     if (this.view.current instanceof WorkspaceContents) {
> ```

Add TODO: Split Editor and remove this disable code.

---

*[src/model/workspace_contents.js, line 41 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/236#-KrBw6taXQJcJcnXHK3A:-KrBw6tbhSQkPZPjtWE9:b-zbboyi) ([raw file](https://github.com/google/blockly-devtools/blob/893d5c46f61fab62eeabe1f3adf19913de31012f/src/model/workspace_contents.js#L41)):*
> ```JavaScript
>     super(workspaceContentsName, PREFIXES.WORKSPACE_CONTENTS);
> 
>     // TODO(#201): Create wrapper class for contents and config object instead
> ```

Remove this TODO.  I don't think it is relevant  going forward in this direction.
I've close #201.

---

*[src/view/workspace_editor_view.js, line 422 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/236#-KrBwRAIfw7NK8RLr3eH:-KrBwRAIfw7NK8RLr3eI:b-qp0ozm) ([raw file](https://github.com/google/blockly-devtools/blob/893d5c46f61fab62eeabe1f3adf19913de31012f/src/view/workspace_editor_view.js#L422)):*
> ```JavaScript
>    */
>   refreshWorkspaceInfo() {
>     if (this.current instanceof WorkspaceContents) {
> ```

TODO(#226): Split this editor into two views.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/236#-:-KrBx9e5NUDoIA-w-ufO:b-fjruft)*
<!-- Sent from Reviewable.io -->

AnmAtAnm reviewed (approved):
LGTM after updating TODO comments.

chynu committed:
Added/removed TODOs

chynu commented:
Review status: 0 of 13 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/controller/workspace_controller.js, line 240 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/236#-KrBvtuGfvybkdn6IOYE:-KrBypG0IJauEQ5bAqja:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/893d5c46f61fab62eeabe1f3adf19913de31012f/src/controller/workspace_controller.js#L240)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Add TODO: Split Editor and remove this disable code.
</blockquote></details>

Done.

---

*[src/model/workspace_contents.js, line 41 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/236#-KrBw6taXQJcJcnXHK3A:-KrByWUkrOZ_M0MiUXcB:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/893d5c46f61fab62eeabe1f3adf19913de31012f/src/model/workspace_contents.js#L41)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Remove this TODO.  I don't think it is relevant  going forward in this direction.
I've close #201.
</blockquote></details>

Done.

---

*[src/view/workspace_editor_view.js, line 422 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/236#-KrBwRAIfw7NK8RLr3eH:-KrByoMNemiZH041WZ2I:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/893d5c46f61fab62eeabe1f3adf19913de31012f/src/view/workspace_editor_view.js#L422)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

TODO(#226): Split this editor into two views.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/236)*
<!-- Sent from Reviewable.io -->
